dali/python/backend_impl.cc

```diff
@@ -273,35 +273,32 @@ void ExposeTensorList(py::module &m) { // NOLINT
       ----------
       )code")
     .def("as_array", [](TensorList<CPUBackend> &t) -> py::array {
+          void* raw_mutable_data;
+          std::string format;
+          size_t type_size;
+
           if (t.size() == 0) {
-            std::string format;
-            size_t type_size;
-            if (IsValidType(t.type())) {
-              format = FormatStrFromType(t.type());
-              type_size = t.type().size();
-            } else {
-              // Default is float
-              format = py::format_descriptor<float>::format();
-              type_size = sizeof(float);
-            }
-            // Based on default numpy empty array shape and strides
-            std::vector<ssize_t> shape(1, 0);
-            std::vector<ssize_t> strides(1, type_size);
-            return py::array(py::buffer_info(nullptr,
-                                             type_size,
-                                             std::move(format),
-                                             shape.size(),
-                                             shape,
-                                             strides));
+            raw_mutable_data = nullptr;
```

Comment:
initialize `raw_mutable_data = nullptr` in line 276 and here just do `if (t.size() > 0)` for what is now the `else`

Reply:
Done